compiler/rustc_resolve/src/late.rs

```diff
@@ -522,7 +522,15 @@ impl<'a: 'ast, 'ast> Visitor<'ast> for LateResolutionVisitor<'a, '_, 'ast> {
         let rib_kind = match fn_kind {
             // Bail if there's no body.
             FnKind::Fn(.., None) => return visit::walk_fn(self, fn_kind, sp),
-            FnKind::Fn(FnCtxt::Free | FnCtxt::Foreign, ..) => FnItemRibKind,
+            // Foreign functions should never have blocks. If one is
+            // invalidly present, we ignore it and only look at the
+            // other parts of the function.
+            FnKind::Fn(FnCtxt::Foreign, _, sig, ..) => {
```

Comment:
Could you merge the previous branch into this new one? They do the same thing.

I had a few doubts on whether we needed to resolve patterns in fn parameters. We don't since they are not allowed in empty & foreign fns. Could you add a comment explaining that?

Reply:
Done!